esp/src/eclwatch/FileBloomsWidget.js

```diff
@@ -0,0 +1,198 @@
+define([
+    "dojo/_base/declare",
+    "dojo/_base/lang",
+    "dojo/i18n",
+    "dojo/i18n!./nls/hpcc",
+    "dojo/_base/array",
+    "dojo/on",
+    "dojo/dom",
+    "dojo/dom-class",
+    "dojo/dom-construct",
+
+    "dijit/registry",
+    "dijit/form/Button",
+    "dijit/ToolbarSeparator",
+    "dijit/Dialog",
+    "dijit/form/TextBox",
+
+    "dgrid/tree",
+    "dgrid/selector",
+
+    "hpcc/GridDetailsWidget",
+    "src/WsDfu",
+    "src/WsESDLConfig",
+    "src/Utility",
+    "hpcc/DelayLoadWidget",
+    "src/ESPUtil",
+    "hpcc/TargetSelectWidget",
+    "hpcc/DelayLoadWidget",
+], function (declare, lang, i18n, nlsHPCC, arrayUtil, on, dom, domClass, domConstruct,
+    registry, Button, ToolbarSeparator, Dialog, TextBox,
+    tree, selector,
+    GridDetailsWidget, WsDfu, WsESDLConfig, Utility, DelayLoadWidget, ESPUtil, TargetSelectWidget, DelayLoadWidget) {
+    return declare("FileBloomsWidget", [GridDetailsWidget, ESPUtil.FormHelper], {
+        i18n: nlsHPCC,
+
+        gridTitle: nlsHPCC.title_Blooms,
+        idProperty: "__hpcc_id",
+
+        init: function (params) {
+            var context = this;
+            if (this.inherited(arguments))
+                return;
+
+            this._refreshActionState();
+            this.refreshGrid();
+
+            ESPUtil.MonitorVisibility(this.gridTab, function (visibility) {
+                if (visibility) {
+                    context.refreshGrid();
+                }
+            });
```

Comment:
I suspect there is no need to refresh on tab selection (does it ever change once the file has been written)?